pkg/kubelet/images/image_manager_test.go

```diff
@@ -509,3 +523,62 @@ func TestEvalCRIPullErr(t *testing.T) {
 		})
 	}
 }
+
+func TestPullAndListImageWithRuntimeClassInImageCriApiFeatureGate(t *testing.T) {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "test_pod",
+			Namespace:       "test-ns",
+			UID:             "bar",
+			ResourceVersion: "42",
+			Annotations: map[string]string{
+				"kubernetes.io/runtimehandler": "handler_name",
+			},
+		}}
+	c := pullerTestCase{ // pull missing image
+		testName:       "test pull and list image with pod annotations",
+		containerImage: "missing_image",
+		policy:         v1.PullIfNotPresent,
+		inspectErr:     nil,
+		pullerErr:      nil,
+		expected: []pullerExpects{
+			{[]string{"GetImageRef", "PullImage"}, nil, true, true},
+		}}
+
+	useSerializedEnv := true
+	t.Run(c.testName, func(t *testing.T) {
+		featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.RuntimeClassInImageCriAPI, true)()
```

Comment:
I think you need a defer here.  At the end of this test, it will set the toggle back to previous setting.  No need fo rline 551.  

I also expect this will fail the linter as we check that defer is required for these feature gates.

Reply:
I usually see cases where one uses a test table with feature off and feature on.  

I would expect to see a case where feature is off and another test case where feature is on.